classes/shaman.lua

```diff
@@ -77,7 +73,8 @@ local RollingThunderHandler = {
         local saoOption = SAO:GetOverlayOptions(324);
         local hasSAO = not saoOption or type(saoOption[lightningShieldStacks]) == "nil" or saoOption[lightningShieldStacks];
         if (hasSAO) then
-            local scale = 0.5 + 0.1 * (lightningShieldStacks - 6);
+            local scale = 0.5 + 0.1 * (lightningShieldStacks - 6); -- 60%, 70%, 80% for Season of Discovery
+            if SAO:IsCata() then local scale = 0.4 + 0.1 * (lightningShieldStacks - 5); end -- 50%, 60%, 70%, 80% for Cataclysm

```

Comment:
Because you redefine `scale` as `local`, its scope will be contained in the if/then/end statement. In essence, this code does nothing. Try this out in the game chat:
```lua
/run local a=1; if true then local a=2 end; print("a == "..a);
```
Also, because the scale computation happens in different places, there is a risk of getting it wrong if someday we change the equation.

I would suggest to declare a local function and use it, for example
```lua
local function getFulminationScale(stacks)
    if SAO.IsSoD() then
        return 0.5 + 0.1 * (stacks - 6); -- 60%, 70%, 80% for Season of Discovery
    elseif SAO.IsCata() then
        return 0.4 + 0.1 * (stacks - 5); -- 50%, 60%, 70%, 80% for Cataclysm
    else
        SAO:Debug(Module, "Fulmination is not supported for current project");
        return 1;
    end
end
```
and then
```lua
if (hasSAO) then
    local scale = getFulminationScale(lightningShieldStacks);
    local pulse = lightningShieldStacks == 9 or nil;
    SAO:ActivateOverlay(lightningShieldStacks, 324, SAO.TexName["fulmination"], "Top", scale, 255, 255, 255, pulse, pulse);
```
...and also call `getFulminationScale` for other code that uses the scale factor.

PS. the `Module` variable is not defined yet, simply define it at the beginning of the file, for example
```lua
local AddonName, SAO = ...
local Module = "shaman"
```

Reply:
Updated in 773860f39b174ce23bab992e81d957455314d387.